src/config/user_profile.cpp

```diff
@@ -53,6 +58,17 @@ profile_pic UserProfile::get_profile_pic() const {
 }
 
 void UserProfile::set_profile_pic(std::string_view url, std::span<const unsigned char> key) {
+    auto current_url = data["p"].string_view_or("");
+    auto current_key_str = data["q"].string_view_or("");
+    std::span<const unsigned char> current_key{
+            reinterpret_cast<const unsigned char*>(current_key_str.data()), current_key_str.size()};
+
+    bool changed = (current_url != url) || (current_key.size() != key.size()) ||
+                   !std::equal(current_key.begin(), current_key.end(), key.begin());
```

Comment:
This is a bit easier to read if you can the span to a string_view, rather than vice versa:
```suggestion
    std::string_view new_key_str{reinterpret_cast<const char*>(key.data()), key.size()};

    bool changed = (current_url != url) || (current_key_str != new_key_str);
```

Reply:
Updated